packages/contracts-bedrock/scripts/upgrades/holocene/DeployUpgrade.s.sol

```diff
@@ -0,0 +1,224 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+// Forge
+import { console2 as console } from "forge-std/console2.sol";
+
+// Scripts
+import { Deployer } from "scripts/deploy/Deployer.sol";
+import { DeployUtils } from "scripts/libraries/DeployUtils.sol";
+
+// Utils
+import "src/dispute/lib/Types.sol";
+
+// Interfaces
+import { ISystemConfig } from "src/L1/interfaces/ISystemConfig.sol";
+import {
+    IFaultDisputeGame,
+    IBigStepper,
+    IAnchorStateRegistry,
+    IDelayedWETH
+} from "src/dispute/interfaces/IFaultDisputeGame.sol";
+import { IPermissionedDisputeGame } from "src/dispute/interfaces/IPermissionedDisputeGame.sol";
+import { IMIPS, IPreimageOracle } from "src/cannon/interfaces/IMIPS.sol";
+
+/// @title DeployUpgrade
+/// @notice A deployment script for smart contract upgrades surrounding the Holocene hardfork.
+contract DeployUpgrade is Deployer {
+    /// @dev The entrypoint to the deployment script.
+    function deploy(address _preimageOracle, address _anchorStateRegistry, address _delayedWETH) public {
+        // Shim the existing contracts that this upgrade is dependent on.
+        shim(_preimageOracle, _anchorStateRegistry, _delayedWETH);
```

Comment:
We also need to allow the user to optionally deploy a DelayedWETH proxy if they are on permissioned and want to move to permissionless

Reply:
I think easiest way to do this is that if the provided `_delayedWETH` is `address(0)` then deploy a proxy for them